views/components/delete_button.erb

```diff
@@ -1,6 +1,10 @@
+<% url = (defined?(url) && url) ? url : request.path %>
 <% csrf_url = (defined?(csrf_url) && csrf_url) ? csrf_url : url %>
 <% text = (defined?(text) && text) ? text : "Delete" %>
 <% confirmation = (defined?(confirmation) && confirmation) ? confirmation : nil %>
+<% confirmation_message = (defined?(confirmation_message) && confirmation_message) ? confirmation_message : nil %>
+<% redirect = (defined?(redirect) && redirect) ? redirect : request.path %>
+<% method = (defined?(method) && method) ? method : "DELETE" %>
```

Comment:
This code works, but it's more accidental than deliberate. In the code `url = defined?(url)` , `defined?(url)` always returns `'local-variable'`, since its on the RHS of the assignment, and the local variable is on the LHS.  The `defined?` calls could be eliminated.  That leaves you with `url = (url ? url : request.path)`.  The idiomatic way to do that in ruby is using `||=`:

```suggestion
<% 
  url ||= request.path
  csrf_url ||= url
  text ||= "Delete"
  confirmation ||= nil
  confirmation_message ||= nil
  redirect ||= request.path
  method ||= "DELETE"
%>
```

Reply:
I added `defined?(url)` because it was failing when we didn't provide this parameter to the component while rendering. It was sad to find out it worked by chance. Fortunately, it did work 😄 I tested the new method, and it also works. However, we use the old syntax in all components. I'll merge this PR as is, and create another one for refactoring the input variables of all components.

Reply:
https://github.com/ubicloud/ubicloud/pull/2200